### ISSUE TITLE
`coordinateTransformations` generated for 0.4 are scale-only

### ISSUE THREAD

**Issue by d-v-b** [ome/ome-zarr-py#403]
The most common methods of image downsampling result in a translation of the downsampled image, but this code for generating `coordinateTransformations` metadata only returns scale transformations, which will be incorrect for almost all multiscale pyramids.

https://github.com/ome/ome-zarr-py/blob/56f72b06d4912ba5156fe54f913d19df895b9e9e/ome_zarr/format.py#L260-L271

Suggested fix:  generate translation transforms

**Comment by will-moore** [comment]
Thanks for this @d-v-b.
Let me just clarify on how to generate the translations here. Looking at the spec 0.4: "If translation is given it MUST be listed after scale to ensure that it is given in physical coordinates".

So, if we have an image that has a pixel size of 1 microns then the 'scale' for each resolution would look like this (given a zoom factor of 2 between resolutions):

```
dataset0: {"scale": [1, 1]}
dataset1: {"scale": [2, 2]}
dataset2: {"scale": [4, 4]}
dataset3: {"scale": [8, 8]}
```
When mapping to physical coordinates, I guess the `translation` needed after scaling depends on where the 'anchor' is when you're scaling. I'm assuming that this is the centre of the pixel at `[0, 0]`. So after scaling then the top-left of that pixel at 2 x its original size will be at `-0.5, 0.5` microns. So then we need to translate `0.5, 0.5`, and so on..

```
dataset0: {"translation": [0, 0]}
dataset1: {"translation": [0.5, 0.5]}
dataset2: {"translation": [1.5, 1.5]}
dataset3: {"translation": [3.5, 3.5]}
```

So the translation for each resolution will be:

`(physical-pixel-size-at-resolution-N - physical-pixel-size-at-resolution-0) / 2`.

That seems to be what https://github.com/thewtex/ngff-zarr is doing when it generates translations.

**Comment by d-v-b** [comment]
yep that's basically it for the most common types of downsampling methods.
